server/consumer.go

```diff
@@ -116,6 +116,33 @@ type SequenceInfo struct {
 type CreateConsumerRequest struct {
 	Stream string         `json:"stream_name"`
 	Config ConsumerConfig `json:"config"`
+	Action ConsumerAction `json:"action"`
+}
+
+type ConsumerAction int
+
+const (
+	ActionCreateOrUpdate ConsumerAction = iota
+	ActionUpdate
+	ActionCreate
+)
+
+func (action *ConsumerAction) UnmarshalJSON(b []byte) error {
```

Comment:
Better to design these ones exactly like the many others we already have, they all follow the same design

https://github.com/nats-io/nats-server/blob/860c481f0f19f9d39fe744d53502c6f7d0746f59/server/store.go#L291-L335

Reply:
Agree. Aligned the implementation.